packages/hosted-form-v2/src/iframe-content/hosted-input-manual-order-payment-handler.ts

```diff
@@ -44,15 +44,40 @@ export default class HostedInputManualOrderPaymentHandler {
         }
 
         try {
-            await this._manualOrderPaymentRequestSender.submitPayment(
+            const response = await this._manualOrderPaymentRequestSender.submitPayment(
                 data,
                 values,
                 this._inputStorage.getNonce(),
             );
 
-            this._eventPoster.post({
-                type: HostedInputEventType.SubmitManualOrderSucceeded,
-            });
+            const isFailure =
+                get(response.body, 'type') === 'failure' && isString(get(response.body, 'code'));
+            const isSuccess = get(response.body, 'type') === 'success';
+            const isError = get(response.body, 'type') === 'error';
+
+            if (isFailure) {
+                this._eventPoster.post({
+                    type: HostedInputEventType.SubmitManualOrderFailed,
+                    payload: {
+                        error: { code: get(response.body, 'code') },
+                    },
+                });
+            }
+
+            if (isError) {
+                this._eventPoster.post({
+                    type: HostedInputEventType.SubmitManualOrderFailed,
+                    payload: {
+                        error: { code: get(response.body, 'type') },
+                    },
+                });
+            }
+
+            if (isSuccess) {
+                this._eventPoster.post({
+                    type: HostedInputEventType.SubmitManualOrderSucceeded,
+                });
+            }
```

Comment:
Are we sure that failure, error, success can happen in one go?

Reply:
No, cannot happen in one go, however we would get only one type at a time. Updated it to use `if.. else if` instead.